lib/addons/core-addon/index.ts

```diff
@@ -43,27 +47,44 @@ export class CoreAddOn implements ClusterAddOn {
         // Create a service account if user provides namespace and service account
         let serviceAccountRoleArn: string | undefined = undefined;
         let serviceAccount: ServiceAccount | undefined = undefined;
+        let saNamespace: string | undefined = undefined;
+
+        saNamespace = DEFAULT_NAMESPACE;
+        if (this.coreAddOnProps?.namespace) {
+            saNamespace = this.coreAddOnProps.namespace;
+        }
 
         if (this.coreAddOnProps?.policyDocumentProvider) {
             const policyDoc = this.coreAddOnProps.policyDocumentProvider(clusterInfo.cluster.stack.partition);
             serviceAccount  = createServiceAccount(clusterInfo.cluster, this.coreAddOnProps.saName,
-                DEFAULT_NAMESPACE, policyDoc);
+                saNamespace, policyDoc);
             serviceAccountRoleArn = serviceAccount.role.roleArn;
 
         }
-        const cfnaddon = new CfnAddon(clusterInfo.cluster.stack, this.coreAddOnProps.addOnName + "-addOn", {
-            addonName: this.coreAddOnProps.addOnName,
-            addonVersion: this.coreAddOnProps.version,
-            clusterName: clusterInfo.cluster.clusterName,
-            serviceAccountRoleArn: serviceAccountRoleArn,
-            resolveConflicts: "OVERWRITE"
-        });
-        
-        if (serviceAccount) {
-            cfnaddon.node.addDependency(serviceAccount);
+        if (this.coreAddOnProps.version) {
```

Comment:
Let's  not duplicate this code just because version has to be defined in one case. 
All we do is add one field to the addon props if version is to be passed. You can define properties as a separate variable and just add an attribute if needed.

You can also drop a property from json, e.g. https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/cluster-providers/generic-cluster-provider.ts#L327

Reply:
This is fixed. Thanks for the suggestion on this.